### PR TITLE
Added community rules, login route changes, updates to community routes

### PR DESCRIPTION
**Pull request by BradyMitch**
# Description

This PR includes the following proposed change(s):

- Server-side parts of HWP-301: Added community rules endpoints
- Updated community PATCH and DELETE operations to check for creator of community
- Added try catch to login route
- Added missing responses to login route
- Removed cookieAuth from swagger as it wasn't actually doing anything. Refresh tokens are stored in cookies.

## Requires

Add 'Requires Attention' label if appropriate.

- [ ] Requires an update to the .env file
- [ ] Requires adding a file listed in the .gitignore
- [ ] Requires an opinion or answer to a question, see comments.
- [x] Requires other attention, see below.

If there are any requirements for the developer to make after this PR is merged please list them here.

Recommended you remove the .mongo/data files. If not, communities you have already created will not work as expected.
@mattsiel @dbarkowsky Updated the login route so it shouldn't respond with any 500 errors.
@ZachBourque Communities now store user.name in the creator field instead of user.id. Rules, description, and creator are now required in the community schema

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

Used swagger to test changed endpoints.

For pull requests that are not just a simple fix, please check that the branch works on your local machine.

- [ ] Tested by Zach.
- [ ] Tested by Brandon.
- [ ] Tested by Brady.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] If attention is required by the developer such as updating the .env file, these requirements have been listed.

### CODE REVIEW

File: app/server/models/community.model.js [hunk 1]
```diff
@@ -42,24 +42,30 @@
...
 
 const mongoose = require("mongoose");
 
 const Community = new mongoose.Schema(
   {
     title: { type: String, required: true, unique: true },
-    description: { type: String },
-    creator: { type: mongoose.Schema.Types.ObjectId, ref: "User" },
+    description: { type: String, required: true },
+    creator: { type: String },
```
Comment (ZachBourque) [inline app/server/models/community.model.js:61]: How come description is required but creator isn't?

Comment (BradyMitch) [reply app/server/models/community.model.js:61]: My bad. Thanks for catching that

**Review by ZachBourque — APPROVED** [review]

**Review by dbarkowsky — APPROVED** [review]
Looks good. Thanks for addressing that so fast.

**Comment by BradyMitch** [comment]
> Looks good. Thanks for addressing that so fast.

Thanks I'm not 100% sure it will fix all the bugs you found as I don't have a way to set a spoofed cookie, but there is extra error handling that might catch it. Let me know how the tests go.

**Pull request merged**